spring-integration-core/src/main/java/org/springframework/integration/gateway/MessagingGatewaySupport.java

```diff
@@ -406,6 +406,13 @@ protected void onInit() {
 			}
 			this.messageConverter.setBeanFactory(beanFactory);
 		}
+		long endpointsDefaultTimeout = getIntegrationProperties().getEndpointsDefaultTimeout();
+		if (!this.requestTimeoutSet) {
+			this.messagingTemplate.setSendTimeout(endpointsDefaultTimeout);
```

Comment:
Is this intentional? Previously, the send timeout still defaulted to -1. Based on the docs, I assume yes. It makes sense to apply it, but I wonder if two properties are needed?

Reply:
It was like this in the ctor:
```
template.setSendTimeout(IntegrationContextUtils.DEFAULT_TIMEOUT);
```

Yes, it was `-1` before `6.1`.

Not sure what you are asking about two props, though...

Reply:
I was just wondering if we need a default send timeout as well as a default reply timeout property; since they are two different things; the send timeout is, for example, if a bounded `QueueChannel` is full whereas the reply timeout is for a request/reply.

Reply:
I see what you mean, but from end-user waiting perspective it looks the same, therefore I believe just a single default property is enough.
You can always override per endpoint and per property if there is a different requirements for send and receive.